sdk/examples/xo_java/XoHandler.java

```diff
@@ -0,0 +1,447 @@
+      player1, player2,
+      signer, action, space
+    );
+
+    storeGameData(gameName, updatedGameData, stateStore);
+  }
+
+  /**
+   * Helper function to get signer public key from transaction header.
+   */
+  private String getSignerPubkey(TpProcessRequest transactionRequest) throws InternalError {
+    try {
+      TransactionHeader header = TransactionHeader.parseFrom(transactionRequest.getHeader());
+      return header.getSignerPubkey();
+    } catch (InvalidProtocolBufferException e) {
+      throw new InternalError("Protocol Buffer Error: " + e.toString());
+    }
+  }
+
+  /**
+   * Helper function to retrieve game name, action, and space from transaction request.
+   */
+  private String[] getUnpackedTransaction(TpProcessRequest transactionRequest) {
+    String[] transactionData = {"", "", ""};
+    String[] payload =  csvDecodeData(transactionRequest.getPayload().toStringUtf8());
+    System.arraycopy(payload, 0, transactionData, 0, payload.length);
+    return transactionData;
+  }
+
+  /**
+   * Helper function to retrieve the board, state, player1, and player2 from state store.
+   */
+  private String[] getStateData(String gameName, State stateStore)
+      throws InternalError, InvalidTransactionException {
+    String[] stateData = {"", "", "", ""};
+    String address = makeGameAddress(gameName);
+
+    String stateEntry = getJsonStringByAddress(address, stateStore);
+    System.out.println("StateEntry is: " + stateEntry);
+    if (stateEntry.length() != 0) {
+      Map<String, String> mapNamesToCsv = getMapNamesToCsv(stateEntry);
+      String dataCsv = mapNamesToCsv.get(gameName);
+      String[] decodedStateData = csvDecodeData(dataCsv);
+      System.arraycopy(decodedStateData, 0, stateData, 0, decodedStateData.length);
+    }
+    return stateData;
+  }
+
+  /**
+   * Helper function to retrieve a JSON string containing the state data for a given address from
+   * the state store.
+   */
+  private String getJsonStringByAddress(String address, State stateStore)
+      throws InternalError, InvalidTransactionException {
+    Map<String, ByteString>  stateEntries = stateStore.get(Collections.singletonList(address));
+    return stateEntries.get(address).toStringUtf8();
+  }
+
+  /**
+   * Helper function to get a map of names to CSV game data from a JSON string.
+   */
+  private Map<String, String> getMapNamesToCsv(String jsonString) {
+    Type type = new TypeToken<Map<String, String>>(){}.getType();
+    GsonBuilder builder = new GsonBuilder();
+    Gson gson = builder.create();
+    Map<String, String> map = gson.fromJson(jsonString, type);
+    if (map != null) {
+      return map;
+    } else {
+      return new HashMap<>();
+    }
+  }
+
+  /**
+   * Helper function to store state data.
+   */
+  private void storeGameData(String gameName, String[] gameData, State stateStore)
+      throws InternalError, InvalidTransactionException {
+    //TODO: Store as an array rather than map
```

Comment:
Methods involving JSON will have to be changed from maps to arrays to ensure a consistent ordering of game names that hash to the same address.